src/sql/workbench/parts/dashboard/widgets/properties/propertiesWidget.component.ts

```diff
@@ -111,16 +138,103 @@ export class PropertiesWidgetComponent extends DashboardWidget implements IDashb
+			const assignProperty = {};
+			let propertyObject = this.getValueOrDefault<string>(this.endpoints[i], EndpointConstants.hyperlink.toString(), '--' || '--');
+
+			assignProperty['displayName'] = endpoint.serviceName;
+			assignProperty['value'] = propertyObject;
+			assignProperty['hyperlink'] = this.isHyperlink(endpoint.serviceName);
+			this.properties.push(<DisplayProperty>assignProperty);
+		}
+
+		if (this._hasInit) {
+			this._changeRef.detectChanges();
+		}
+	}
+
+	private getEndpoints(): void {
+		const endpointArray = this._connection.serverInfo.options[EndpointConstants.clusterEndpoints.toString()];
+
+		// TO DO: Once the endpoints data is moved to DMV, update this logic to replace hardcoded values
+		// with the data from DMV.
+		if (endpointArray && endpointArray.length > 0) {
+			// iterate over endpoints and display them
+			this.endpoints = [];
+			for (let i = 0; i < endpointArray.length; i++) {
+				const ep = endpointArray[i];
+				const assignProperty = {};
+				assignProperty['serviceName'] = ep.serviceName;
+				assignProperty['ipAddress'] = ep.ipAddress;
+				assignProperty['port'] = ep.port;
+				assignProperty['hyperlink'] = ep.ipAddress + ':' + ep.port;
+				this.endpoints.push(<Endpoint>assignProperty);
+			}
+
+			// add grafana and kibana endpoints from the management proxy endpoint
+			const managementProxyEp = endpointArray.find(e => e.serviceName === EndpointConstants.managementproxy.toString());
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('metrics') > -1)) {
+				this.endpoints.push(this.addCustomEndpoint(managementProxyEp, 'Grafana Dashboard', '/grafana'));
+			}
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('log') > -1)) {
+				this.endpoints.push(this.addCustomEndpoint(managementProxyEp, 'Kibana Dashboard', '/kibana'));
+			}
+
+			// add spark and yarn endpoints form the gateway endpoint
+			const gatewayEndpoint = endpointArray.find(e => e.serviceName === EndpointConstants.gateway.toString());
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('spark') > -1)) {
+				this.endpoints.push(this.addCustomEndpoint(gatewayEndpoint, 'Spark History', '/gateway/default/sparkhistory'));
+			}
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('log') > -1)) {
+				this.endpoints.push(this.addCustomEndpoint(gatewayEndpoint, 'Yarn History', '/gateway/default/yarn'));
+			}
+		}
+	}
+
+	private isHyperlink(serviceName: string): boolean {
+		return !(serviceName.toLowerCase().indexOf(EndpointConstants.appproxy.toString()) > -1 || serviceName.toLowerCase().indexOf(EndpointConstants.controller.toString()) > -1
+			|| serviceName.toLowerCase().indexOf(EndpointConstants.gateway.toString()) > -1 || serviceName.toLowerCase().indexOf(EndpointConstants.managementproxy.toString()) > -1);
+	}
+
+	private addCustomEndpoint(parentEndpoint: IEndpoint, serviceName: string, serivceUrl: string): Endpoint {
+		const endpoint = {};
+		if (parentEndpoint) {
```

Comment:
This should probably be defined as an Endpoint to begin with and given appropriate default values.